src/NexusMods.App.Cli/Types/IpcHandlers/NxmIpcProtocolHandler.cs

```diff
@@ -80,6 +86,39 @@ public async Task Handle(string url, CancellationToken cancel)
                     _logger.LogWarning("Download failed: User is not logged in");
                 }
                 break;
+            case NXMCollectionUrl collectionUrl:
+                if (userInfo is not null)
+                {
+                    var domain = GameDomain.From(collectionUrl.Game);
+                    var nexusModsLibrary = _serviceProvider.GetRequiredService<NexusModsLibrary>();
+                    var library = _serviceProvider.GetRequiredService<ILibraryService>();
+                    var gameRegistry = _serviceProvider.GetRequiredService<IGameRegistry>();
+                    if (!gameRegistry.InstalledGames.TryGetFirst(g => g.Game.Domain == domain, out var game))
+                        return;
+                    
+                    var syncService = _serviceProvider.GetRequiredService<ISynchronizerService>();
+                    if (!syncService.TryGetLastAppliedLoadout(game, out var loadout))
+                        return;
+                    
+                    var temporaryFileManager = _serviceProvider.GetRequiredService<TemporaryFileManager>();
+                    await using var destination = temporaryFileManager.CreateFile();
+
+                    var downloadJob = nexusModsLibrary.CreateCollectionDownloadJob(destination, collectionUrl.Collection.Slug, collectionUrl.Revision,
+                        CancellationToken.None
+                    );
+        
+                    var libraryFile = await library.AddDownload(downloadJob);
+        
+                    if (!libraryFile.TryGetAsNexusModsCollectionLibraryFile(out var collectionFile))
+                        throw new InvalidOperationException("The library file is not a NexusModsCollectionLibraryFile");
+
+                    var installJob = await InstallCollectionJob.Create(_serviceProvider, loadout, collectionFile);
```

Comment:
Should this code be in here? I think the same for the normal nxm download above. I would prefer this just call a method of some service that is in charge of downloads or library.
At the least I would prefer this be extracted to a local method to keep the switch case more readable.